ggshield/iac/output/iac_text_output_handler.py

```diff
@@ -1,25 +1,54 @@
 import shutil
+from collections import namedtuple
 from io import StringIO
 from pathlib import Path
-from typing import ClassVar, List
+from typing import ClassVar, Dict, Generator, List, Optional
 
 from pygitguardian.iac_models import IaCFileResult, IaCVulnerability
 
 from ggshield.core.text_utils import (
     STYLE,
     Line,
     clip_long_line,
+    file_diff_info,
     file_info,
     format_text,
     get_offset,
     get_padding,
+    pluralize,
 )
 from ggshield.core.utils import Filemode, get_lines_from_content
 from ggshield.iac.collection.iac_diff_scan_collection import IaCDiffScanCollection
 from ggshield.iac.collection.iac_path_scan_collection import IaCPathScanCollection
+from ggshield.iac.iac_scan_models import IaCDiffScanEntities
+from ggshield.iac.output.iac_output_handler import IaCOutputHandler
 from ggshield.scan import File
 
-from .iac_output_handler import IaCOutputHandler
+
+GroupedIncidents = namedtuple(
+    "GroupedIncidents", ["name", "new", "unchanged", "deleted"]
+)
+
+
+def group_incidents_by_filename(
+    incidents: IaCDiffScanEntities,
+) -> Generator[GroupedIncidents, None, None]:
+    keys = []
+    attrs = ["new", "unchanged", "deleted"]
+    attrs_val: Dict[str, dict] = dict()
+    for attr in attrs:
+        attrs_val[attr] = dict()
```

Comment:
You could instanciate your dictionnary `attrs_vals` explicitely. eg:
```py
attrs_vals = {"new": {}, "unchanged": {}, "deleted": {}}
```

And then iterate on it.